routers/api/v1/repo/issue.go

```diff
@@ -55,12 +61,36 @@ func ListIssues(ctx *context.APIContext) {
 		isClosed = util.OptionalBoolFalse
 	}
 
-	issues, err := models.Issues(&models.IssuesOptions{
-		RepoIDs:  []int64{ctx.Repo.Repository.ID},
-		Page:     ctx.QueryInt("page"),
-		PageSize: setting.UI.IssuePagingNum,
-		IsClosed: isClosed,
-	})
+	// Define issues var
+	var issues []*models.Issue
+
+	// Check for search
+	keyword := strings.Trim(ctx.Query("q"), " ")
+	if bytes.Contains([]byte(keyword), []byte{0x00}) {
+		keyword = ""
+	}
+	var issueIDs []int64
+	var err error
+	if len(keyword) > 0 {
+		issueIDs, err = indexer.SearchIssuesByKeyword(ctx.Repo.Repository.ID, keyword)
+
+		// Didn't found anything
+		if len(issueIDs) == 0 {
+			issues = []*models.Issue{}
+		}
+	}
+
+	// Show the results if we either dont have a keyword or the issues found by said keyword are > 0
```

Comment:
nit: This comment is not needed, its obvious from code

Reply:
Oh yeah you're right, I removed it.